ckanext/harvest/logic/schema.py

```diff
@@ -12,39 +12,82 @@
                                    )
 from ckan.logic.converters import convert_to_extras, convert_from_extras
 
-from ckan.lib.navl.validators import (ignore_missing,
-                                      not_empty,
-                                      ignore,
-                                      if_empty_same_as,
-                                      )
-
-from ckanext.harvest.logic.validators import (harvest_source_url_validator,
-                                              harvest_source_type_exists,
-                                              harvest_source_config_validator,
-                                              harvest_source_extra_validator,
-                                              harvest_source_frequency_exists,
-                                              dataset_type_exists,
-                                              harvest_source_convert_from_config,
-                                              harvest_source_id_exists,
-                                              harvest_job_exists,
-                                              harvest_object_extras_validator,
-                                              )
+from ckanext.harvest.logic.validators import (
+    harvest_source_url_validator,
+    harvest_source_type_exists,
+    harvest_source_config_validator,
+    harvest_source_extra_validator,
+    harvest_source_frequency_exists,
+    dataset_type_exists,
+    harvest_source_convert_from_config,
+    harvest_source_id_exists,
+    harvest_job_exists,
+    harvest_object_extras_validator,
+)
+ckan_version = int(h.ckan_version().split(".")[1])
```

Comment:
@TomeCirun rather than conditionally loading things in different part of the module let's define our own cross-compatible version in this module and use it on the schemas below

```python
def unicode_safe(value):
    if ckan_version >= 9:
         from ckan.lib.navl.validators import unicode_safe as core_unicode_safe
         return core_unicode_safe(value)
    else:
         return six.text_type(value)
```

And then we just use it on the schemas

```python
schema['id'] = [ignore_missing, unicode_safe]
```